sklearnex/tests/test_memory_usage.py

```diff
@@ -79,28 +79,30 @@ def dataframe_f(x, y):
 ]
 
 
-def gen_clsf_data():
+def gen_clsf_data(estimator_name):
     data, label = make_classification(
         n_samples=2000, n_features=50, random_state=777)
-    return data, label, \
-        data.size * data.dtype.itemsize + label.size * label.dtype.itemsize
+    data_size = data.size * data.dtype.itemsize + label.size * label.dtype.itemsize
+    if estimator_name == 'SVC':
+        data_size += 2048 * label.size * label.dtype.itemsize
+    return data, label, data_size
 
 
 def _kfold_function_template(estimator, data_transform_function):
     tracemalloc.start()
 
-    x, y, data_memory_size = gen_clsf_data()
+    x, y, data_memory_size = gen_clsf_data(estimator.__name__)
     kf = KFold(n_splits=10)
     x, y = data_transform_function(x, y)
 
-    mem_before, _ = tracemalloc.get_traced_memory()
     for train_index, test_index in kf.split(x):
         if isinstance(x, np.ndarray):
             x_train, x_test = x[train_index], x[test_index]
-            y_train, y_test = y[train_index], y[test_index]
+            y_train = y[train_index]
         elif isinstance(x, pd.core.frame.DataFrame):
             x_train, x_test = x.iloc[train_index], x.iloc[test_index]
-            y_train, y_test = y.iloc[train_index], y.iloc[test_index]
+            y_train = y.iloc[train_index]
+        mem_before, _ = tracemalloc.get_traced_memory()
```

Comment:
Memory tracing should be outside of `kf.split` cycle. For every loop, memory leak grows and becomes more visible. Previous cases, when full input data leaked, were easy to find, but future leaks might be more sneaky so this cycle is used to find them